### PR TITLE
Fix ExoPlayer: avoid stretching of HLS when aspect ratio changes

### PR DESCRIPTION
**Pull request by brianephraim**
The ExoPlayer implementation was not properly handling anamorphic HLS playlists, i.e. videos that change size or aspect ratio during playback.  iOS already handled this complexity fine.

Here is an HLS that exhibits this anamorphic behavior:  
https://hls.ted.com/talks/93.m3u8

## before fix
https://hls.ted.com/talks/93.m3u8
stretched-video

## after fix
fixed-stretch-video

Thanks for opening a PR!
Since this is a volunteer project and is very active, anything you can do to reduce the amount of time needed to review and merge your PR is appreciated.
The following steps will help get your PR merged quickly:

#### Update the documentation
If you've added new functionality, update the README.md with an entry for your prop or event.
The entry should be inserted in alphabetical order.

#### Update the changelog
After you open the PR, update the CHANGELOG.md file with an entry pointing to your PR.

#### Provide an example of how to test the change
If the PR requires special testing setup provide all the relevant instructions and files. This may include a sample video file or URL, configuration, or setup steps.

#### Focus the PR on only one area
Testing multiple features takes longer than isolated changes and if there is a bug in one feature, prevents the other parts of your PR from getting merged until it gets fixed.
If you're touching multiple different areas that aren't related, break the changes up into multiple PRs.

#### Describe the changes
Add a note describing what your PR does. If there is a change to the behavior of the code, explain why it needs to be updated.

### CODE REVIEW

**Comment by nbennink** [comment]
Thanks it works for me. Don't forget to remove the (now useless) comment and variable in the code.

**Comment by NrqhEIcsRi004** [comment]
Good PR :)

**Comment by NrqhEIcsRi004** [comment]
Can a ratio change be used for a callback?

**Comment by egor-bukharov** [comment]
Hi! The PR is pending for 2 years, does this fix work now?

**Comment by yegorius** [comment]
We have recently hit this bug in production and it affects a lot of our customers. Can someone please merge it?

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java [hunk 1]
```diff
@@ -201,13 +201,8 @@ public void onCues(List<Cue> cues) {
 
         @Override
         public void onVideoSizeChanged(int width, int height, int unappliedRotationDegrees, float pixelWidthHeightRatio) {
-            boolean isInitialRatio = layout.getAspectRatio() == 0;
             layout.setAspectRatio(height == 0 ? 1 : (width * pixelWidthHeightRatio) / height);
-
-            // React native workaround for measuring and layout on initial load.
-            if (isInitialRatio) {
```
Comment (freeboub) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.java:208]: With this patch you will have strange behavior.
If you are playing content with ABR, onVideoSizeChanged will be called on every time playlist change.
you need to check the aspect ratio have a real change,
something like this:
abs(oldAspectRatio - newAspectRatio) > aspectRatioChangeThreesold

**Pull request closed without merging**